go/apps/api/openapi/openapi.yaml

```diff
@@ -2083,9 +1688,57 @@ components:
         - interval
         - amount
       additionalProperties: false
+    KeyCreditsRequest:
+      type: object
+      description: Credit configuration and remaining balance for this key.
+      properties:
+        remaining:
+          type: integer
+          format: int64
+          minimum: -1 # -1 represents unlimited
+          maximum: 9223372036854775807
+          description: Number of credits remaining (-1 for unlimited).
+          example: 1000
+        refill:
+          "$ref": "#/components/schemas/KeyCreditsRefillRequest"
+      required:
+        - remaining
+      additionalProperties: false
+    KeyCreditsRefillRequest:
+      type: object
+      properties:
+        interval:
+          type: string
+          enum:
+            - daily
+            - monthly
+          description: How often credits are automatically refilled.
+          example: daily
+        amount:
+          type: integer
+          format: int64
+          minimum: 1
+          maximum: 1000000
+          description: Number of credits added during each refill.
+          example: 100
+        refillDay:
+          type: integer
+          minimum: 1
+          maximum: 31
+          description: Day of month for monthly refills (1-31).
+          example: 1
+      required:
+        - interval
+        - amount
+      additionalProperties: false
     KeyResponse:
       type: object
       properties:
+        apiId:
```

Comment:
this was omitted by design, cause in the future we might have a key be part of multiple apis or even no api.

Keys belong to a key_auth, and in the future we will allow you to attach a key_auth to a gateway for example, or maybe multiple key_auths even. so there will not be a strict n:1 relation between keys and apis